pandas/core/arrays/datetimes.py

```diff
@@ -236,83 +236,52 @@ class DatetimeArray(dtl.DatetimeLikeArrayMixin,
     _dtype = None  # type: Union[np.dtype, DatetimeTZDtype]
     _freq = None
 
-    def __init__(self, values, dtype=_NS_DTYPE, freq=None, copy=False):
-        if isinstance(values, (ABCSeries, ABCIndexClass)):
-            values = values._values
-
-        if isinstance(values, type(self)):
-            # validation
-            dtz = getattr(dtype, 'tz', None)
-            if dtz and values.tz is None:
-                dtype = DatetimeTZDtype(tz=dtype.tz)
-            elif dtz and values.tz:
-                if not timezones.tz_compare(dtz, values.tz):
-                    msg = (
-                        "Timezone of the array and 'dtype' do not match. "
-                        "'{}' != '{}'"
-                    )
-                    raise TypeError(msg.format(dtz, values.tz))
-            elif values.tz:
-                dtype = values.dtype
-            # freq = validate_values_freq(values, freq)
-            if freq is None:
-                freq = values.freq
-            values = values._data
+    def __init__(self, values, dtype=None, freq=None, copy=False):
+        if freq == "infer":
+            raise ValueError(
+                "Frequency inference not allowed in DatetimeArray.__init__. "
+                "Use 'pd.array()' instead.")
 
-        if not isinstance(values, np.ndarray):
-            msg = (
-                "Unexpected type '{}'. 'values' must be a DatetimeArray "
+        if not hasattr(values, "dtype"):
+            # e.g. list
+            raise ValueError(
+                "Unexpected type '{vals}'. 'values' must be a DatetimeArray "
                 "ndarray, or Series or Index containing one of those."
-            )
-            raise ValueError(msg.format(type(values).__name__))
+                .format(vals=type(values).__name__))
 
-        if values.dtype == 'i8':
-            # for compat with datetime/timedelta/period shared methods,
-            #  we can sometimes get here with int64 values.  These represent
-            #  nanosecond UTC (or tz-naive) unix timestamps
-            values = values.view(_NS_DTYPE)
-
-        if values.dtype != _NS_DTYPE:
-            msg = (
+        if values.dtype == np.bool_:
```

Comment:
why do u only check for book here? shouldn’t this. e inside _from_sequeneve?

Reply:
well, the good news is that I'm not hallucinating non-existent test cases.  the relevant test was for `to_datetime(False, errors="coerce")`, so I'm moving this check back up to `__init__`.  To be clear, I don't think the three checks in `__init__` should be there at all, but for the moment I'm trying to minimize behavior changes and just get the tests passing.

Reply:
It's not because `to_datetime` has that behaviour that there should be a check for that here?